examples/http/sonicwall_ipv4_dhcp_leases.js

```diff
@@ -0,0 +1,146 @@
+    config.auth = "basic";
+    config.username = D.device.username();
+    config.password = D.device.password();
+    D.device.http.post(config, processResponse(d));
+    return d.promise;
+}
+
+/**
+ * Retrieves DHCP leases info
+ * @returns {promise} The promise for retrieving DHCP leases info
+ */
+function getDHCPLeases() {
+    var d = D.q.defer();
+    var config = generateConfig("/api/sonicos/reporting/dhcp-server/ipv4/leases/statistic");
+    D.device.http.get(config, processResponse(d));
+    return d.promise;
+}
+
+function getDisplayValue(value) {
+  return (value === undefined || value === null || value === '') ? 'N/A' : value
+}
+
+/**
+ * Extracts the variables from DHCP leases info
+ * @param {object} dhcpLeasesBody Body of the DHCP leases API response
+ */
+function extractVariables(dhcpLeasesBody) {
+    var dhcpLeases = JSON.parse(dhcpLeasesBody)
+
+    var current = getDisplayValue(dhcpLeases.current);
+    var avaliable_dynamic = getDisplayValue(dhcpLeases.avaliable_dynamic);
+    var avaliable_static = getDisplayValue(dhcpLeases.avaliable_static_);
+    var total_active = getDisplayValue(dhcpLeases.total_active);
+    var total_configured = getDisplayValue(dhcpLeases.total_configured);
+
+    var variables = [
+        D.createVariable("current", "Current", current, null, D.valueType.NUMBER),
+        D.createVariable("avaliable_dynamic", "Available Dynamic", avaliable_dynamic, null, D.valueType.NUMBER),
+        D.createVariable("avaliable_static", "Available Static", avaliable_static, null, D.valueType.NUMBER),
+        D.createVariable("total_active", "Total Active", total_active, null, D.valueType.NUMBER),
+        D.createVariable("total_configured", "Total Configured", total_configured, null, D.valueType.NUMBER),
+    ];
+    D.success(variables);
+}
+
+
+/**
+ * @remote_procedure
+ * @label Validate Connection 
+ * @documentation This procedure is used to validate the connection and data retrieval from the SonicWALL device.
+ */
+function validate(){
+    login()
+        .then(getDHCPLeases)
+        .then(function (response) {
+            if (response) {
+                console.info("Data available");
+                D.success();
+            } else {
+                console.error("No data available");
+                D.failure(D.errorType.GENERIC_ERROR);
+            }
+        })
+        .catch(function (err) {
+            console.error(err);
+            D.failure(D.errorType.GENERIC_ERROR);
+        });
+}
+
+/**
+ * @remote_procedure
+ * @label Get Sonicwall Product lifecycle
+ * @documentation This procedure is used to retrieve the product lifecycle information for a SonicWall device. The product lifecycle includes five phases: Last Day Order (LDO), Active Retirement (ARM), One-Year Support Last Day Order, Limited Retirement Mode (LRM), and End of Support (EOS). 
+ * This information is used to monitor the product lifecycle and check when a product is approaching end of support
+ */
```

Comment:
This function is intended to retrieve DHCP leases, not product lifecycle data. The documentation should be updated.